### ISSUE TITLE
When to use `SOME_DATA.toU8a(true)`

### ISSUE THREAD

**Issue by mustermeiszer** [polkadot-js/api#3762]
Sorry, to open an issue again. 

But I am wondering what are the implications of using `toU8a(true)`? Are there some docs to read, as the description of the interface was not really clear to me.

My questions would be: 
- What changes when using this option?
- Are these changes equivalent to every type?
- Are there types, on which this option have no effect at all?

Thanks again for your help!

**Comment by jacogr** [comment]
For most types it has no effect. It is only used when you know the underlying SCALE encoding and need to override. 

An example is extrinsics, the first byte is the start of the compact-encoded length of the whole extrinsic. However is some cases you want to have a specific encoding, eg. for the signed payload (what generates the transaction signature), the length portion is not signed. So in that case this flag will not add the length, but also end up with an encoding that cannot be passed through in normal cases. 

TL;DR it is a very special encoding override flag

**Comment by mustermeiszer** [comment]
Ah okay. I get for example the following error when I do not use this option in this example:

```
fromApi.createType('(Vec<(AccountId, ProxyType)>, Balance)', scaleOldProxies);
```

Error is: 
```
Error: createType((Vec<(AccountId, ProxyType)>, Balance)):: decodeU8a: failed at 0x010484cfba018307… on : Vec<(AccountId,ProxyType)>:: decodeU8a: failed at 0x000063472573db04… on : (AccountId,ProxyType):: decodeU8a: failed at 0x000063472573db04… on : AccountId:: Invalid AccountId provided, expected 32 bytes, found 16
```

The two scale values differ in the first byte of the array. One is 50 bytes long (the working one) and one is 51 bytes long (the failing one). The first byte is a 1.

Can you tell me, why this is failing?

This is really specific but I have no clue where to look at the reasoning for why this is different for this particular type. You can also lead me to the impl if this is faster for you.

**Comment by jacogr** [comment]
Options are SCALE encoded in the following way -

- the first byte is 0 when the value is None
- the first byte is 1, when the value is Some
- remainder is the SCALE-encoded wrapped data

So in the case of Option, the `isBare` flag will forgo the first byte (generally this would be useful when converting to hex, when the it is the the actual data or null, no preceding indicator byte needed)

**Comment by mustermeiszer** [comment]
Thank you! I will go with hex-encoded values then, as I don't know the type.
